src/librustc_lint/unused.rs

```diff
@@ -355,7 +388,50 @@ impl UnusedParens {
                         left_pos.map(|s| s >= value.span.lo()).unwrap_or(false),
                         right_pos.map(|s| s <= value.span.hi()).unwrap_or(false),
                     );
-                    Self::remove_outer_parens(cx, value.span, &expr_text, msg, keep_space);
+                    Self::emit_unused_parens(cx, value.span, &expr_text, ctx.into(), keep_space);
+                }
+            }
+            ast::ExprKind::Block(ref inner, None)
+                if inner.rules == ast::BlockCheckMode::Default =>
+            {
+                // emit a warning under the following conditions:
+                //
+                // - the block does not have a label
+                // - the block is not `unsafe`
+                // - the block contains exactly one expression (do not lint `{ expr; }`)
+                // - `followed_by_block` is true and the internal expr may contain a `{`
+                // - the block is not multiline
```

Comment:
What's the rationale for this?

Reply:
because of the current formatting style:

```rust
match a {
    Some(v) => really_long_expr,
    // ...
}
```
gets formatted as

```rust
match a {
    Some(v) => {
        really_long_expr
    }
    // ...
}
```

Reply:
Alright, please extend the comment with a brief mention of formatting style.